domain/access/state/permission.go

```diff
@@ -686,7 +735,9 @@ FROM    cloud
 WHERE   name = $M.grant_on
 `
 	case corepermission.Offer:
-		return errors.NotImplementedf("db permission support for offers")
+		// TODO(aflynn): Add referential integrity check for offers once they
```

Comment:
Is there a jira card for this in the offer domain epic?

Reply:
Added task https://warthogs.atlassian.net/browse/JUJU-6575